engine/mongodb/mongodb-core/src/test/java/com/torodb/mongodb/core/MongoDbBsonTypesTest.java

```diff
@@ -0,0 +1,209 @@
+            {"REGEX_WITH_OPTIONS", new DefaultBsonRegex("gimy", "asd")},
+            {"DB_POINTER", new DefaultBsonDbPointer("asd", new IntBasedBsonObjectId(1, 1, 1, 1))},
+            {"JAVA_SCRIPT", new DefaultBsonJavaScript("alert(\"hello\");")},
+            {"DEPRECATED", new StringBsonDeprecated("Deprecable me")},
+            //{"JAVA_SCRIPT_WITH_SCOPE", new DefaultBsonJavaScriptWithCode("alert(\"hello\");", new SingleEntryBsonDocument("class", new StringBsonString(".main-menu")))},
+            {"INT32", PrimitiveBsonInt32.newInstance(55)},
+            {"TIMESTAMP", new DefaultBsonTimestamp(1482000000, 1)},
+            {"INT64", PrimitiveBsonInt64.newInstance(1525155)},
+            {"DECIMAL128", new LongsBsonDecimal128(1L,1L)},
+            {"DECIMAL128_NaN", new LongsBsonDecimal128(8935141660703064064L,1L)},
+            {"DECIMAL128_INFINITE", new LongsBsonDecimal128(8646911284551352320L,1L)},
+            {"MIN", SimpleBsonMin.getInstance()},
+            {"MAX", SimpleBsonMax.getInstance()}
+    });
+
+    Arrays.asList(BsonType.values()).forEach(
+            type -> {
+              if (type != BsonType.JAVA_SCRIPT_WITH_SCOPE) {
+                assertTrue(type + " type is never tested", allTests.stream().anyMatch(
+                        toTest -> type.getValueClass().isAssignableFrom(toTest[1].getClass())
+                        )
+                );
+              }
+            }
+    );
+
+    return allTests;
+  }
+
+  @Parameterized.Parameter(0)
+  public String collName;
+
+
+  @Parameterized.Parameter(1)
+  public BsonValue value;
+
+  @Test
+  public void test() {
+    List<BsonDocument> docs = new ArrayList<>();
+    docs.add(new SingleEntryBsonDocument("number", value));
+
+    baseWriteTest(docs, collName);
+  }
+
+  private void baseWriteTest(List<BsonDocument> docs, String collName) {
+
+    writeTransaction = bundle.getExternalInterface().getMongodServer().openConnection().openWriteTransaction();
+
+    //Prepare and do the insertion
+
+    InsertCommand.InsertArgument insertArgument = new InsertCommand.InsertArgument.Builder(collName)
+            .addDocuments(docs)
+            .build();
+    Status<InsertCommand.InsertResult> insertResultStatus =
+            writeTransaction.execute(request, InsertCommand.INSTANCE, insertArgument);
+
+    //Prepare and do the retrieval
+
+    FindCommand.FindArgument findArgument = new FindCommand.FindArgument.Builder()
+            .setCollection(collName)
+            .build();
+    Status<FindCommand.FindResult> findResultStatus =
+            writeTransaction.execute(request, FindCommand.INSTANCE, findArgument);
+
+    writeTransaction.close();
+
+
+    //Asserting that the status of both operations are right
+
+    assertTrue(insertResultStatus.getResult().getWriteErrors().isEmpty());
+    assertTrue(insertResultStatus.isOk()?"ok":insertResultStatus.getErrorMsg() + " in collection " + collName, insertResultStatus.isOk());
```

Comment:
Because if it was ok, the getErrorMsg was throwing an Exception. It is ugly but I didn't realize it because it is a message in a test class